### PR TITLE
Add React with Next.js example

### LINKED ISSUE #18: Add Example of Using Bootstrap with React JS frontend library
Hello , I would like to add example of some demo react JS pages showing how we can utilize Bootstrap in React .

### PR DESCRIPTION
**Pull request by kyletsang**
Adds an example using React and Next.js

Closes #18

StackBlitz preview: https://stackblitz.com/github/kyletsang/examples/tree/react-nextjs/react-nextjs?file=src%2Fpages%2Findex.tsx

### CODE REVIEW

**Comment by XhmikosR** [comment]
@kyletsang thanks for the PR! I pushed a patch with some fixes. Now, only lint is left to fix.

File: react-nextjs/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "compilerOptions": {
+    "target": "es5",
```
Comment (XhmikosR) [inline react-nextjs/tsconfig.json:3]: Maybe this is too low for nowadays?

Comment (kyletsang) [reply react-nextjs/tsconfig.json]: Good catch.  Looks like `create-next-app` has an old default for TS projects.  This should've been bumped higher with nextjs 13.

**Comment by XhmikosR** [comment]
Thanks! Only lint is left to tackle, but we can also disable the rules if you think it's better.

**Comment by kyletsang** [comment]
> Thanks! Only lint is left to tackle, but we can also disable the rules if you think it's better.

I'll take care of this in a few hours 😄

**Comment by XhmikosR** [comment]
@kyletsang any reason why you went with react-bootstrap instead of vanilla bootstrap? I'm not familiar with the tooling here, but if there are no issues maybe we could stick to vanilla bootstrap?

**Comment by kyletsang** [comment]
> @kyletsang any reason why you went with react-bootstrap instead of vanilla bootstrap? I'm not familiar with the tooling here, but if there are no issues maybe we could stick to vanilla bootstrap?

This is mostly to resolve potential incompatibilities with the JS plugin components like Offcanvas, popper-related components etc.  React manages its own DOM mutations, so you aren't suppose to mutate it yourself otherwise it leads to issues like crashing. That aside, RB also handles some other things such as accessibility, setting class names based on props, etc so users don't have to wire those their own base components.

There's some more information in this link regarding DOM manipulation:
https://react.dev/learn/manipulating-the-dom-with-refs#best-practices-for-dom-manipulation-with-refs

**Comment by XhmikosR** [comment]
Thanks for the explanation!

I guess it's fine to go with this approach, then :)

**Pull request merged**